### DISCUSSION TITLE
EF - SqlException: Invalid column name 'PostID1'.

### DISCUSSION THREAD

**Discussion by BrunoBafilli** [balta-io/forum#270] [Tira Dúvidas] [answered]
Boa noite a todo.

Venho repetido o modulo EF pela quarta vez para garantir que não estou cometendo algum erro, porem, de todas as aulas que assiste acredito que o Balta não fala sobre o relacionamento entre POST E TAGS no sentido de adicionar isso no banco de dados, eu posso estar eganado, mas acredito que não.

Eu gostaria de saber como faço para adicionar um POST com todas as TAGS relacionadas a ele no banco, sendo que, ao adicionar um post com varias tags relacionadas, teoricamente a minha tabela de relacionamento (POSTTAGS) deveria ser preenchida com todas TAGS e POSTS relacionados correto?

Por algum motivo isso não está acontecendo, a minha tabela PostTags está ficando vazia, não está criando o relacionamento.

**Código**

```
using ConsoleApp2.Data;
using ConsoleApp2.Model;



BlogContext blogContext = new BlogContext();

blogContext.Tags.Add(new Tag() { Nome = "C#"});
blogContext.Tags.Add(new Tag() { Nome = "Javascript"});

blogContext.Posts.Add(new Post() { Titulo = "Postado de teste", Tags = blogContext.Tags.ToList() });

blogContext.SaveChanges();
```

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @BrunoBafilli, como vai?
Segue o código:

```csharp
var tags = new List<Tag> { new Tag() { Nome = "C#"} };
var post = new Post { ..., Tags = tags };

context.Posts.Add(post);
context.SaveChanges();
```
